libs/wingc/src/type_check.rs

```diff
@@ -3220,6 +3219,18 @@ impl<'a> TypeChecker<'a> {
 		first_expected_type
 	}
 
+	pub fn validate_type_is_stringable(&mut self, actual_type: TypeRef, span: &impl Spanned) -> TypeRef {
+		if actual_type.is_stringable() {
+			return actual_type;
+		}
+
+		let message = format!("Expected type to be stringable, but got \"{actual_type}\" instead");
+		let hint = "str, num, bool, json, and enums are stringable".to_string();
+		self.spanned_error_with_hints(span, message, vec![hint]);
+
+		self.types.string()
+	}
```

Comment:
nit: The return type doesn't seem to be used, is it needed?

```suggestion
	pub fn validate_type_is_stringable(&mut self, actual_type: TypeRef, span: &impl Spanned) {
		if !actual_type.is_stringable() {
		  let message = format!("Expected type to be stringable, but got \"{actual_type}\" instead");
		  let hint = "str, num, bool, json, and enums are stringable".to_string();
		  self.spanned_error_with_hints(span, message, vec![hint]);
		}
	}
```

(or if you want to avoid nesting, an early `return;`)

Reply:
👍  Fixed